{{cookiecutter.project_slug}}/clients/vue3/src/views/Login.vue

```diff
@@ -15,11 +16,12 @@
         :errors="form.password.errors"
         @blur="form.password.validate()"
         type="password"
+        id="test-password"
         label="Password:"
         placeholder="Password"
       />
 
-      <button type="submit">Login</button>
+      <button id="test-submit" type="submit">Login</button>
```

Comment:
Not sure if there are opinions but years ago this was how we made it clear that the automated tests expected something in the HTML to exist. Not sure if there are better/cleaner ways now. Would love opinions on this.

Reply:
I think this could be a good standard, though I might suggest coming up with a custom `data-` attribute, as opposed to using IDs, so that IDs, if used, can be semantic. I'd also support just using IDs without the `test` prefix. In any case, it's going to be something the developers have to be aware of and maintain.

Reply:
While adding an id does work, I recommend adding data-cy attributes instead, as per https://docs.cypress.io/guides/references/best-practices#How-It-Works.

That way it is clear that these data-attributes are there for cypress to target.

Reply:
I think the `data-*` attribute might be the best route to go here. I know I used `id`'s when doing it before, but Cypress' docs ([here](https://docs.cypress.io/guides/references/best-practices)) say the best practice is to use the `data-*` attribute so that you're using a selector that isn't also _potentially_ tied to CSS and/or JS.